build.rs

```diff
@@ -757,8 +762,36 @@ fn configure(interpreter_config: &InterpreterConfig) -> Result<()> {
         _ => {}
     }
 
-    if interpreter_config.shared {
-        println!("cargo:rustc-cfg=Py_SHARED");
+    if env::var_os("CARGO_FEATURE_AUTO_INITIALIZE").is_some() {
+        ensure!(
+            interpreter_config.shared,
+            "The `auto-initialize` feature is enabled, however your python installation only \
```

Comment:
This document is nice, but I'm a bit doubtful that users want to read such a long message (this means that I'm embarrassingly lazy to read errors:sweat:).

Reply:
Thanks, this and the comments below were very helpful. I've pushed a new commit which makes the build error much shorter.